packages/@react-aria/focus/src/useFocusRing.ts

```diff
@@ -41,12 +41,18 @@ export function useFocusRing(props: FocusRingProps = {}): FocusRingAria {
     within
   } = props;
   let state = useRef({
-    isFocused: autoFocus,
-    isFocusVisible: isFocusVisible()
+    isFocused: false,
+    isFocusVisible: autoFocus || isFocusVisible()
   }).current;
   let [isFocusVisibleState, setFocusVisible] = useState(() => state.isFocused && state.isFocusVisible);
 
   let updateState = () => setFocusVisible(state.isFocused && state.isFocusVisible);
+
+  let onFocus = () => {
+    state.isFocused = true;
+    updateState();
+  };
```

Comment:
Is this necessary? I think `onFocusChange` should cover this.

Reply:
I don't think it's needed either, at least, when i removed it locally with the other changes, things appeared to work